src/pages/en/core-concepts/astro-components.mdx

```diff
@@ -11,9 +11,9 @@ i18nReady: true
 
 Astro component syntax is a superset of HTML. The syntax was [designed to feel familiar to anyone with experience writing HTML or JSX](#differences-between-astro-and-jsx), and adds support for including components and JavaScript expressions. You can spot an Astro component by its file extension: `.astro`.
 
-Astro components are extremely flexible. Often, an Astro component will contain some **reusable UI on the page**, like a header or a profile card. At other times, an Astro component may contain a smaller snippet of HTML, like a collection of common `<meta>` tags that make SEO easy to work with. Astro components can even contain an entire page layout.
+Astro components are extremely flexible. Often, an Astro component will contain some **reusable UI on the page**, like a header or a profile card. At other times, an Astro component may contain a smaller snippet of HTML, like a collection of common `<meta>` tags that make SEO easy to work with. It can even contain an entire page layout.
 
-The most important thing to know about Astro components is that they **render to HTML during your build.** Even if you run JavaScript code inside of your components, it will all run ahead of time, stripped from the final page that you send to your users. The result is a faster site, with zero JavaScript footprint added by default.
+The most important thing to know about Astro components is that they **render to HTML ahead of time**. It could be during the build step or on the server on demand, depending on how you have configured your project. Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default. 
```

Comment:
Not sure about the wording here, saying "Astro components render to HTML ahead of time [....] on demand in the server" feels kinda contradictory to me? How about changing this a bit?

```suggestion
The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in SSR. Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

Reply:
That sounds better 👍 

I guess we're allowed to use the word "SSR" here?

Reply:
Great question! I feel like SSR is such a common buzzword nowadays that most people will understand we are talking about server-side rendering, although I can see the argument for using the full term instead, I'll let @sarah11918 weight on this.

Reply:
We're allowed to say it! Often if it's at the very top of a page like this, we might spell it out in full and then also put (SSR) in brackets after. And, I would probably choose to link to the SSR page from it.

Reply:
Agree!

```suggestion
The most important thing to know about Astro components is that they **render to HTML**, be it ahead of time in static builds or on-demand in [server-side rendering (SSR)](/en/guides/server-side-rendering/). Even if you run JavaScript code inside of your components, all of it will be stripped from the final page sent to your users' browsers. The result is a faster site, with zero JavaScript footprint added by default.
```

Reply:
Sounds good!